### ISSUE TITLE
ModelCheckpoint  need monitor average loss

### ISSUE THREAD

**Issue by Johnson-yue** [Lightning-AI/pytorch-lightning#20652]
### Description & Motivation

I know ModelCheckpoint can monitor like "train_loss" , "val_loss" , when the value is min  and  "every_n_train_steps" is true.  but I want to cache "train_loss" or "val_loss" from "every_n_train_steps" to next "every_n_train_steps" ,when cache loss is lower than before,  then save model. 
Example:
every_n_train_steps = 50, monitor = "train_loss"

from step=50 to step=100, if average "train_loss"  is lower than step=[0,50] then save model
